pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -626,15 +647,20 @@ func (c *VSphereController) createCSISecret(
 		return fmt.Errorf("cloud config %s/%s does not contain key %q", cloudConfigNamespace, cloudConfig.Name, cloudConfig.Key)
 	}
 
-	var cfg vsphere.VSphereConfig
-	err = gcfg.ReadStringInto(&cfg, cfgString)
+	// Update this to also support YAML.
+	//var cfg vsphere.VSphereConfig
+	//err = gcfg.ReadStringInto(&cfg, cfgString)
```

Comment:
can we drop this commented code or are we leaving it in for historical reasons?

Reply:
These will be cleaned up shortly.  Still working on merging conflicts and fixing.